src/partition/repart.c

```diff
@@ -3321,6 +3373,31 @@ static int do_copy_files(Partition *p, const char *root) {
         return 0;
 }
 
+static int do_make_subvolumes(Partition *p, const char *root) {
+        int r;
+
+        assert(p);
+        assert(root);
+
+        STRV_FOREACH(d, p->make_subvolumes) {
+                _cleanup_free_ char *path = NULL;
+
+                path = path_join(root, *d);
+                if (!path)
+                        return log_oom();
+
+                r = mkdir_parents_safe(root, path, 0755, UID_INVALID, GID_INVALID, 0);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parent directories of '%s' in file system: %m", path);
+
+                r = btrfs_subvol_make(path);
```

Comment:
as mentioned above, I think we should fall back to making dirs in case of ERRNO_IS_NOT_SUPPORTED() here (and debug log about this)